src/numerics/petsc_matrix.C

```diff
@@ -769,7 +769,7 @@ void PetscMatrix<T>::print_personal(std::ostream & os) const
       // of the file internally.  Since print_personal() takes a reference to
       // an ostream, we have to do an extra step...  print_personal() should probably
       // have a version that takes a string to get rid of this problem.
-      ierr = PetscViewerASCIIOpen( this->comm().get(),
+      ierr = PetscViewerASCIIOpen( PetscObjectComm((PetscObject)_mat),
```

Comment:
Hmm... if there is there some difference between the communicator of the PetscMatrix and the underlying Mat, I believe we will run into other issues as well, no? Can you explain more about the situation where this arises?

Reply:
@fdkong will we run into other issues due to communicator differences?

Reply:
`Viewer` and `Mat` have to use the same communicator. We have PETSc code like

`PetscCheckSameComm(mat,1,viewer,2);`

Reply:
@fdkong I agree that `Viewer` and `Mat` should should use the same Communicator! The issue I see is that there should be no difference between `this->comm().get()` (where `this` refers to the `PetscMatrix`) and `PetscObjectComm((PetscObject)_mat` under most normal circumstances. Do you agree? Otherwise it means that someone has changed either the` PetscMatrix` or  `Mat` communicator but not updated the other one about the change...

Reply:
It seems that I should keep this line as it originally was then? I am not sure which is better..

Reply:
Logically I think it should be the same. If it actually "fixes" a problem, then there is a larger issue that we would need to look into and analyze, which is how the Communicator object of the `PetscMatrix` and its underlying `_mat` can possibly be different.

Reply:
Thanks @jwpeterson . IIRC the error I was having actually came from the `MatView()`. Let me try my case without this line of change and see if I still see the error.

Reply:
I do not have time to verify it at this moment. From my understanding (I could be totally wrong), PETSc can not use the same communicator as libmesh or other packages. If PETSc can use  the same comm, I will be puzzled on

`MPI_Isend(comm, .., tag.., ).`

PETSc could generate the same tag as libmesh, and then how we make sure we do not mess up this kind of communication.

Reply:
OK, I understood now. PETSc will retrieve an inner comm from libmesh comm if it exists, otherwise, PETSc will duplicate a communicator and imbed it into libmesh-comm.  So PETSc will always have its own comm. So `MPI_Isend` will be fine.  

`PetscCheckSameComm` will still return true as long as libmesh-comm and PETSc-comm share the same group.

So the bottom line is that  you could leave this code alone

Reply:
Thanks @fdkong for looking into this! I checked my case again and it seems this line of change is not necessary. I am reverting it back to what it was.

Reply:
Yes, your case will not touch this line of code at all. It is why you did not see anything. But I do not think we need to change this code